autoendpoint/src/routes/webpush.rs

```diff
@@ -12,15 +13,31 @@ use actix_web::HttpResponse;
 pub async fn webpush_route(
     notification: Notification,
     routers: Routers,
+    state: Data<ServerState>,
 ) -> ApiResult<HttpResponse> {
     let router = routers.get(
         RouterType::from_str(&notification.subscription.user.router_type)
             .map_err(|_| ApiErrorKind::InvalidRouterType)?,
     );
 
-    let response = router.route_notification(&notification).await?;
-
-    Ok(response.into())
+    match router.route_notification(&notification).await {
+        Ok(v) => Ok(v.into()),
+        Err(e) => {
+            if !e.kind.user_still_valid() {
+                // The user record is no longer valid. We should remove it so that the client
+                // can attempt to recover. We don't care if this succeeds or fails at this point.
+                if drop_user(
+                    notification.subscription.user.uaid,
+                    state.ddb.as_ref(),
+                    &state.metrics,
+                )
+                .await
+                .is_ok()
+                {};
```

Comment:
the if .. {} is a weird read, alternatively:
```suggestion
                let _ = drop_user(
                    notification.subscription.user.uaid,
                    state.ddb.as_ref(),
                    &state.metrics,
                )
                .await;
```